core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -371,32 +371,24 @@ void CudaInternal::fence() const {
 }
 
 void CudaInternal::initialize(cudaStream_t stream, bool manage_stream) {
+  KOKKOS_EXPECTS(!is_initialized());
+
   if (was_finalized)
     Kokkos::abort("Calling Cuda::initialize after Cuda::finalize is illegal\n");
   was_initialized = true;
-  if (is_initialized()) return;
-
-  const bool ok_init = nullptr == m_scratchSpace || nullptr == m_scratchFlags;
 
-  if (ok_init) {
```

Comment:
For the record
https://github.com/kokkos/kokkos/blob/c692a816df60691482950c54bc88ea8ef51e0db8/core/src/Cuda/Kokkos_Cuda_Instance.hpp#L157-L159